api/src/main/java/org/pentaho/platform/api/genericfile/model/IGenericFile.java

```diff
@@ -106,6 +107,21 @@ default boolean isFolder() {
    */
   boolean isCanEdit();
 
+  /**
```

Comment:
Docs could state the order of the folders in the list.
Docs could state that it is possible some of the folders in the list may not exist anymore.